service/common/src/main/java/org/apache/polaris/service/catalog/iceberg/IcebergCatalog.java

```diff
@@ -1328,6 +1383,21 @@ public void doCommit(TableMetadata base, TableMetadata metadata) {
       String newLocation = writeNewMetadataIfRequired(base == null, metadata);
       String oldLocation = base == null ? null : base.metadataFileLocation();
 
+      // TODO: we should not need to do this hack, but there's no other way to modify
+      // currentMetadata / currentMetadataLocation
+      if (updateMetadataOnCommit) {
+        try {
+          tableMetadataField.set(metadata, newLocation);
+          unsafe.putObject(metadata, changesFieldOffset, new ArrayList<MetadataUpdate>());
```

Comment:
If we have to resolve to this kind of trickery in order to make Iceberg code behave as we want it to behave, I wonder if it might be preferable to revamp this class as a whole and directly implement the logic for storing and loading metadata from storage. WDYT?

Reply:
I'm very uneasy about modifying private stuff in method args.

Do I understand this correctly, that you guys talked about this offline and this is still the easiest fix short of rewriting `IcebergCatalog` for direct storage acces?

Reply:
If you're interested in an alternate implementation, please check out #1426 where I refactor the TableOperations / ViewOperations classes in such a way that I could follow up with logic to manage this metadata without reflection!